Code/GraphMol/MolOps.cpp

```diff
@@ -183,62 +184,104 @@ void halogenCleanup(RWMol &mol, Atom *atom) {
+    return true;
+  }
+
+  return false;
+}
+
+int numDativeBonds(const Atom *atom) {
+  int numDatives = 0;
+  auto &mol = atom->getOwningMol();
+  for (auto bond : mol.atomBonds(atom)) {
+    if (bond->getBondType() == Bond::BondType::DATIVE ||
+        bond->getBondType() == Bond::BondType::DATIVEONE ||
+        bond->getBondType() == Bond::BondType::DATIVEL ||
+        bond->getBondType() == Bond::BondType::DATIVER) {
+      ++numDatives;
+    }
+  }
+  return numDatives;
+}
+
 void metalBondCleanup(RWMol &mol, Atom *atom) {
   PRECONDITION(atom, "bad atom in metalBondCleanup");
   // The IUPAC recommendation for ligand->metal coordination bonds is that they
   // be single.  This upsets the RDKit valence model, as seen in CHEBI:26355,
   // heme b.  If the valence of a non-metal atom is above the maximum in the
   // RDKit model, and there are single bonds from it to metal
   // change those bonds to atom->metal dative.
+  // If the atom is bonded to more than 1 metal atom, choose the one
+  // with the fewer dative bonds incident on it, with the canonical
+  // rank of the atoms as a tie-breaker.
 
-  auto isMetal = [](const Atom *a) -> bool {
-    // This is the list of not metal atoms from QueryOps.cpp
-    static const std::set<int> notMetals{0,  1,  2,  5,  6,  7,  8,  9,
-                                         10, 14, 15, 16, 17, 18, 33, 34,
-                                         35, 36, 52, 53, 54, 85, 86};
-    return (notMetals.find(a->getAtomicNum()) == notMetals.end());
-  };
   auto noDative = [](const Atom *a) -> bool {
     static const std::set<int> noD{1, 2, 9, 10};
     return (noD.find(a->getAtomicNum()) != noD.end());
   };
-  if (!isMetal(atom)) {
-    return;
-  }
-  const auto &valens =
-      PeriodicTable::getTable()->getValenceList(atom->getAtomicNum());
-  if (valens.back() != -1) {
-    // the atom can only have specific valences, so leave it.
-    return;
-  }
-  for (auto bond : mol.atomBonds(atom)) {
-    auto otherAtom = bond->getOtherAtom(atom);
-    if (!isMetal(otherAtom) && !noDative(otherAtom)) {
-      auto ev = otherAtom->calcExplicitValence(false);
-      // Check the explicit valence of the non-metal against the allowed
-      // valences of the atom, adjusted by its formal charge.  This means that
-      // N+ is treated the same as C, O+ the same as N.  This allows for,
-      // for example, c1cccc[n+]1-[Fe] to be acceptable and not turned into
-      // c1cccc[n+]1->[Fe].  After all, c1cccc[n+]1-C is ok.  Although this is
-      // a poor example because c1ccccn1->[Fe] appears to be the normal
-      // way that pyridine complexes with transition metals.  Heme b in
-      // CHEBI:26355 is an example of when this is required.
-      int effAtomicNum =
-          otherAtom->getAtomicNum() - otherAtom->getFormalCharge();
-      if (effAtomicNum <= 0) {
-        continue;
+
+  std::vector<unsigned int> ranks(mol.getNumAtoms());
+  RDKit::Canon::rankMolAtoms(mol, ranks, true, true, true);
```

Comment:
Since you're using the defaults, you might as well skip the extra arguments:
```suggestion
  RDKit::Canon::rankMolAtoms(mol, ranks);
```